javatests/arcs/schematests/Arcs.kt

```diff
@@ -0,0 +1,185 @@
+@file:Suppress("EXPERIMENTAL_IS_NOT_ENABLED")
+
+package arcs.schematests
+
+import android.content.Context
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LifecycleRegistry
+import arcs.android.sdk.host.AndroidHost
+import arcs.android.storage.database.AndroidSqliteDatabaseManager
+import arcs.core.allocator.Allocator
+import arcs.core.common.toArcId
+import arcs.core.host.EntityHandleManager
+import arcs.core.host.SchedulerProvider
+import arcs.core.host.toRegistration
+import arcs.core.storage.api.DriverAndKeyConfigurator
+import arcs.jvm.host.ExplicitHostRegistry
+import arcs.jvm.host.JvmSchedulerProvider
+import arcs.jvm.util.JvmTime
+import arcs.sdk.android.storage.ServiceStoreFactory
+import arcs.sdk.android.storage.service.ConnectionFactory
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlin.coroutines.EmptyCoroutineContext
+
+
+/** Container to own the allocator and start the long-running arc. */
+class Arcs(
+    private val context: Context,
+    // A test [ConnectionFactory] can be provided here under test.
+    // In production, leave this parameter as null. Arcs will provide a default implementation.
+    connectionFactory: ConnectionFactory? = null
+) {
+    val schedulerProvider = JvmSchedulerProvider(EmptyCoroutineContext)
+
+
+    private val fakeLifecycleOwner = object : LifecycleOwner {
+        private val lifecycle = LifecycleRegistry(this)
+        override fun getLifecycle() = lifecycle
+    }
+
+
+    val arcHost = ArcHost(
+        context,
+        fakeLifecycleOwner.lifecycle,
+        schedulerProvider,
+        connectionFactory
+    )
+
+    private lateinit var allocator: Allocator
+
+    /**
+     * Start the arc containing the particles that provide the persistence. This should be run
+     * one time per instance, via [startIfNot].
+     */
+    private suspend fun startArc() {
+        /** This should probably live in an initializer somewhere. */
+        val hostRegistry = ExplicitHostRegistry().apply {
+            registerHost(arcHost)
+        }
+
+        allocator = Allocator.create(
```

Comment:
I think we can move this to a helper, and encapsulate in a Dagger module.

Reply:
We shouldn't require our users to use Dagger... but using a helper/builder/etc. makes sense.

Reply:
On the one hand I agree, but on the other we've had explicit requests from folks like Peru to make it so that for example their use of Arcs is just like using Sting/Dagger with which they're already familiar and use frequently, so it's worth considering both.

Reply:
It should be possible to deliver parts of Arcs-things with Dagger, but we should not make dagger a dependency of Arcs and especially not a requirement for using Arcs.

Reply:
Agree strongly with @jasonwyatt. We can include examples showing how to use the creation builders in a Dagger module.

Reply:
I'm not saying Dagger in OSS, if you make a helper, then in G3 the helper can be wrapped in Sting/Dagger.

Reply:
The wrapping is really easy to do and we shouldn't embed any helper class/method in the Sting modules. 

And if we can build a helper method to make things easy, I think we can also "inline" the helper method into implementations if people only use the helper version to do these initializations.